sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/MetricFeedback/MetricFeedback.cs

```diff
@@ -8,44 +8,91 @@
-        /// <exception cref="ArgumentNullException"> <paramref name="dimensionFilter"/> is null. </exception>
-        internal MetricFeedback(string metricId, FeedbackDimensionFilter dimensionFilter)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MetricFeedback"/> class.
+        /// </summary>
+        /// <param name="metricId">The identifier of the metric to which the <see cref="MetricFeedback"/> applies.</param>
+        /// <param name="dimensionKey">
+        /// A key that identifies a set of time series to which the <see cref="MetricFeedback"/> applies.
+        /// If all possible dimension columns are set, this key uniquely identifies a single time series
+        /// for the specified <paramref name="metricId"/>. If only a subset of dimension columns are set, this
+        /// key uniquely identifies a group of time series.
+        /// </param>
+        /// <exception cref="ArgumentNullException"><paramref name="metricId"/> or <paramref name="dimensionKey"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="metricId"/> is empty.</exception>
+        internal MetricFeedback(string metricId, DimensionKey dimensionKey)
         {
             Argument.AssertNotNullOrEmpty(metricId, nameof(metricId));
-            Argument.AssertNotNull(dimensionFilter, nameof(dimensionFilter));
+            Argument.AssertNotNull(dimensionKey, nameof(dimensionKey));
 
             MetricId = metricId;
-            DimensionFilter = dimensionFilter;
+            DimensionKey = dimensionKey;
         }
 
-        /// <summary> The <see cref="MetricFeedbackKind"/> of this feedback.</summary>
+        internal MetricFeedback(MetricFeedbackKind kind, string id, DateTimeOffset? createdTime, string userPrincipal, string metricId, FeedbackDimensionFilter dimensionFilter)
+        {
+            Kind = kind;
+            Id = id;
+            CreatedTime = createdTime;
+            UserPrincipal = userPrincipal;
+            MetricId = metricId;
+            DimensionKey = dimensionFilter.DimensionFilter;
+        }
+
+        /// <summary>
+        /// The kind of feedback.
+        /// </summary>
         [CodeGenMember("FeedbackType")]
         public MetricFeedbackKind Kind { get; internal set; }
 
-        /// <summary> feedback unique id. </summary>
+        /// <summary>
+        /// The unique identifier of the <see cref="MetricFeedback"/>. Set by the service.
+        /// </summary>
         [CodeGenMember("FeedbackId")]
         public string Id { get; }
 
-        /// <summary> feedback created time. </summary>
+        /// <summary>
+        /// Date and time, in UTC, when the <see cref="MetricFeedback"/> was created.
+        /// </summary>
         public DateTimeOffset? CreatedTime { get; }
 
-        /// <summary> user who gives this feedback. </summary>
+        /// <summary>
+        /// The user who created the <see cref="MetricFeedback"/>.
```

Comment:
is it a user alias or email or id or is there something specific worth saying?

Reply:
I'm pretty sure you were not expecting this question to be a huge catch. Thank you.

I always assumed it was just the creator's email, which was true before we added AAD. Checked in our recordings to double check, and I saw an ID being used in a few tests.

After investigating further, it depends on the type of authentication used. I updated the docstring to:
```
/// The user who created the <see cref="MetricFeedback"/>. If <see cref="MetricsAdvisorKeyCredential"/>
/// authentication was used when creating the feedback, this property contains the email address of the
/// creator. If AAD authentication was used instead, the value of this property uniquely identifies the
/// creator's user principal, but its value depends on the type of credential used. For instance, if a
/// <c>ClientSecretCredential</c> is used, it will contain the client ID.
```

This affects other properties we have in other models as well (such as `DataFeed.AdministratorEmails`). Double checked them and the same logic applies. Discussing with other languages right now, but we'll probably need to rename them.

Once again, thank you!

Reply:
wow!!! I am so glad it was useful. :)
Thank you for following up and validating it!! great job!